### ISSUE TITLE
RHESSI fits sum_flag=0 breaks code

### ISSUE THREAD

**Issue by ianan** [sunpy/sunxspex#132]
### Describe the bug

At the moment the code returns an error and breaks when trying to load a RHESSI srm/spec fits file with sum_flag != 1 in the header.  I guess @KriSun95 did this is to avoid issues of multiple detectors that aren't summed, i.e. expecting always to have a single srm and single spectrogram in each fits file. But even if there is a single srm/spec in there (say only using a single RHESSI detector) will get the error/break if sum_flag=0 in the fits header.

Short term solution is just to make sure sum_flag=1 when generating RHESSI srm/spec fits from the hsi_spectrum() in sswidl, whether summed over multiple detectors or "summed" over a single detector.

Longer term might want a smarter way to deal with this? Maybe @settwi might have some ideas given your work on loading in RHESSI spec/srm.

### To Reproduce

Load in a RHESSI spec/srm fits file where have a single detector spectrogram/srm but sum_flag=0 in the header, i.e. https://github.com/ianan/rhessi_spectra/blob/main/mfstats9_fits/20030725_082642_srm_org_d1.fits https://github.com/ianan/rhessi_spectra/blob/main/mfstats9_fits/20030725_082642_spec_org_d1.fits

`from sunkit_spex.sunxspex_fitting.fitter import Fitter`
`spec = Fitter(pha_file="20030725_082642_spec_org_d1.fits", srm_file="20030725_082642_srm_org_d1")`

Gives error/breaks in sunxspex_fitting/instruments.py:662 with warning:
"Apparently spectrum file's SUMFLAG should be one and I don't know what to do otherwise at the moment."

**Comment by settwi** [comment]
I believe OSPEX only allows spectroscopy to be performed when `SUMFLAG=1` (see [here](https://hesperia.gsfc.nasa.gov/ssw/packages/spex/idl/object_spex/spex_hessi_fits2drm.pro)):
```idl
sum_flag = fxpar (hdr, 'SUMFLAG', count=count)
if count gt 0 and sum_flag eq 0 then begin
  if n_elements(str2arr(detused, ' ')) gt 1 then begin   ; check for > 1 det
	  err_code = 1
	  err_msg = 'OSPEX can not handle SRM files with >1 detectors that are not summed (sum_flag=0). Aborting'
	  return
	endif
endif
```

Summing counts from the different RHESSI detectors is kind of fraught so that makes sense.
I guess we could just add a special case if there is only one detector/SRM. OSPEX does that a few lines above.

-WS

**Comment by KriSun95** [comment]
The `spex_hessi_fits2drm.pro` file that @settwi has linked is likely why my error very helpfully reads: 
`"Apparently spectrum file's SUMFLAG should be one and I don't know what to do otherwise at the moment."`. 

Looking at it again, would this not most likely be because OSPEX can't do simultaneous fitting? I.e., it would have no idea what to do if the detectors weren't summed into a single spectrum with only one SRM. Obviously, if this is the _only_ reason why `SUMFLAG` must be 1, then there would be no issue on our side by supporting `SRM files with >1 detectors that are not summed`.

**Comment by settwi** [comment]
Good idea!

**Comment by samaloney** [comment]
Is this still an issue?

**Comment by ianan** [comment]
Yes, still returns error "Cannot perform spectroscopy on un-summed RHESSI data."  This is setup to produce an error in OSPEX if `sum_flag=0` and more than one detector is present.... we just have an error if `sum_flag !=1`

Work around is always set `sum_flag=1` when generating them in RHESSI spectrum object, whether or not you are using more than one detector.

Could make the check a default that could be turned off, i.e. in `sunkit-spex/extern/rhessi.py` 
```python
class RhessiLoader:
    def __init__(self, spectrum_fn, srm_fn, require_sumflag=True):
        self._construction_string = (
            f"RhessiLoader(spectrum_fn={spectrum_fn}, srm_fn={srm_fn})"
        )
        self._systematic_error = 0
        self.load_prepare_spectrum_srm(
            spectrum_fn, srm_fn, require_sumflag=require_sumflag
        )

    def load_prepare_spectrum_srm(self, spectrum_fn, srm_fn, require_sumflag=True):
        self._spectrum = (spec := load_spectrum(spectrum_fn, require_sumflag=require_sumflag))
        self._attenuator_state_info = spec.pop("attenuator_state_info")
        self._srm = (srm := load_srm(srm_fn))
        # ... rest unchanged


def load_spectrum(spec_fn, require_sumflag=True):
    # ... existing code above unchanged
    rate_dat = spec["RATE"]
    if require_sumflag and rate_dat.header.get("SUMFLAG", 1) != 1:
        raise ValueError("Cannot perform spectroscopy on un-summed RHESSI data.")
    # ... rest unchanged
```
This will be moved over to the RHESSILoader sunkit-instruments at some point - make the change before to the legacy version so that it will get carried over?

**Comment by settwi** [comment]
Fitting data that isn’t summed requires simultaneous fitting. We can adjust the behavior in legacy to load in multiple spectra from a single FITS if that’s desired.

**Comment by ianan** [comment]
Not necessarily, it could just be from one detector. That's why OSPEX has a double check of flag and number of spectra.

**Comment by settwi** [comment]
Fair enough. That would be easy to add. For the multi-detector files, that would require some thinking.
